### PR TITLE
Delete a Region after its spectral profile is finished

### PR DESCRIPTION
**Pull request by markccchiang**
This is for issue #299.

While deleting the Region:
1. Set the connection flag is false to interrupt the running job.
2. Wait for the job finished before deleting the Region.

### CODE REVIEW

File: Frame.cc [hunk 1]
```diff
@@ -889,6 +890,10 @@ bool Frame::FillSpectralProfileData(
         if (!region->IsValid()) {
             return false;
         }
+
+        // Increase the spectral profile count
+        region->IncreaseZProfileCount();
```
Comment (pford) [inline Frame.cc]: There are cases where this function returns without decreasing the zprofile count (no profiles or region outside image), so the region's count will never reach zero.  Why doesn't the region inc/dec its own count when its zprofile functions are called?

File: Frame.cc [hunk 2]
```diff
@@ -1252,15 +1261,15 @@ bool Frame::GetPointSpectralData(std::vector<float>& data, int region_id, casaco
                 // start the timer
                 auto t_start = std::chrono::high_resolution_clock::now();
                 // check if region point changed from subimage point
-                if ((region_id == CURSOR_REGION_ID) && (Interrupt(_cursor_xy, subimage_cursor))) {
+                if ((region_id == CURSOR_REGION_ID) && (Interrupt(region_id, _cursor_xy, subimage_cursor))) {
                     return false; // cursor moved
```
Comment (pford) [inline Frame.cc]: Return values for GetPointSpectralData and GetRegionSpectralData are not checked, so if the profile is interrupted it is still sent to the frontend.  I think this profile is no longer needed.

**Comment by pford** [comment]
Overall I think the region having a connection status and waiting for the zprofiles to complete will work.  A couple issues in the implementation.

File: Frame.cc (hunk 1 above)
Comment (markccchiang) [reply Frame.cc]: Not sure is it a good solution. Now I synchronized the zprofile counts for the Frame and the Region objects in the commit 716337aeec2a65e711e07f6543bf34194adc54f8. To make sure the zprofile process is completely done, and then destroy the objects.

**Pull request merged**